src/main/java/prefs/ConfigFile.java

```diff
@@ -0,0 +1,131 @@
+package prefs;
+
+import com.google.gson.*;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import util.HTLog;
+
+import java.io.*;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Optional;
+
+/**
+ * A general class for HubTurbo config files.
+ * Handles serialisation and deserialisation of config objects from and to files.
+ */
+public class ConfigFile {
```

Comment:
This class looks like it's simply handling a general Json conversion, not a Config-specific action.